src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/DirectoryBasedStoreProvider.cs

```diff
@@ -395,5 +402,64 @@ private static void EnsureFilePermissions(FileStream stream, uint userId)
                 }
             }
         }
+
+        internal sealed class UnsupportedDisallowedStore : IStorePal
+        {
+            private readonly bool _readOnly;
+
+            internal UnsupportedDisallowedStore(OpenFlags openFlags)
+            {
+                // ReadOnly is 0x00, so it is implicit unless either ReadWrite or MaxAllowed
+                // was requested.
+                OpenFlags writeFlags = openFlags & (OpenFlags.ReadWrite | OpenFlags.MaxAllowed);
+
+                if (writeFlags == OpenFlags.ReadOnly)
+                {
+                    _readOnly = true;
+                }
+
+                string storePath = GetStorePath(X509Store.DisallowedStoreName);
+
+                if (Directory.Exists(storePath))
+                {
+                    // If it has no files, leave it alone.
+                    foreach (string filePath in Directory.EnumerateFiles(storePath))
```

Comment:
Since the user would have to manually delete the directory (we never will), I'm not sure it's a state worth worrying about.  But if it seems sensible to do so, I can wrap the whole `if` here in a try/catch-IOException-and-ignore-it statement.